validator/entrypoint.sh

```diff
@@ -14,7 +14,13 @@ SIGNER_API_URL=$(get_signer_api_url "${NETWORK}" "${SUPPORTED_NETWORKS}")
 BEACON_API_URL=$(get_beacon_api_url "${NETWORK}" "${SUPPORTED_NETWORKS}" "${CLIENT}")
 MEVBOOST_FLAG=$(get_mevboost_flag "${MEVBOOST_FLAG_KEY}" "${SKIP_MEVBOOST_URL}")
 
-BEACON_API_4000="$(echo "$BEACON_API_URL" | cut -d':' -f1,2):4000"
+# Extract base URL. This assumes BEACON_API_URL will has the following format: http://<domain>:<port> 
+# Example: http://localhost:4000 -> localhost
+BASE_URL="$(echo "$BEACON_API_URL" | cut -d'/' -f3 | cut -d':' -f1)"
+
+# Prepare API endpoints (no http:// prefix && add port)
+BEACON_API_4000="${BASE_URL}:4000"
```

Comment:
Maybe it's better to name both as they name them:

4000 -> `BEACON_RPC_PROVIDER`
3500 -> `BEACON_RPC_GATEWAY_PROVIDER`